lua/mediaplayer/services/youtube/init.lua

```diff
@@ -134,18 +82,127 @@ function SERVICE:GetMetadata( callback )
 		callback(self._metadata)
 
 	else
-
 		local videoId = self:GetYouTubeVideoId()
-		local apiurl = MetadataUrl:format( videoId, APIKey )
+		local videoUrl = "https://www.youtube.com/watch?v="..videoId
 
-		self:Fetch( apiurl,
+		self:Fetch( videoUrl,
+			-- On Success
 			function( body, length, headers, code )
-				OnReceiveMetadata( self, callback, body )
+				local metadata = self:ParseYTMetaDataFromHTML(body, videoId)
+
+				-- html couldn't be parsed
+				if not metadata.title or not isnumber(metadata.duration) then
+					callback(false, "Failed to parse HTML Page for metadata")
+					return
+				end
```

Comment:
In case the parsing code breaks at some point, it'll be good to catch any of the potentially erroring code. You'll want to test these changes, I'm not 100% sure it works.
```suggestion
				local status, metadata = pcall( self.ParseYTMetaDataFromHTML, body, videoId )

				-- html couldn't be parsed
				if not status or not metadata.title or not isnumber(metadata.duration) then
					callback(false, "Failed to parse HTML Page for metadata: " .. metadata)
					return
				end
```